Picard event details: workflow sub-tabs

SimoneSchmidt opened the pull request:
Adds all sub-tabs of the workflow-tab to the event details modal, with both views and logic.
Adds some missing comments.

SimoneSchmidt committed:
Workflow Tab of Event Details finished

Finished logic for Sub-Tab "Workflow Details".
Added views and logic for Sub-Tabs "Operations", "Operation Details", "Errors" and "Error Details".

SimoneSchmidt committed:
comments added

code comments added to html sections for Event Details Comments Tab

SimoneSchmidt committed:
Merge branch 'admin-ui-picard' into picard-workflow-details

Merge to include bug fixes

isabellakutger commented on modules/admin-ui-picard/app/src/components/events/partials/ModalTabsAndPages/EventDetailsWorkflowOperationDetails.js:
```diff
@@ -0,0 +1,176 @@
+        marginRight: "30px",
+        fontWeight: "600",
+        color: "#5d7589"
+    }
```
Maybe put these in a separate file and import it here and in the other files using these consts

isabellakutger commented on modules/admin-ui-picard/app/src/components/events/partials/ModalTabsAndPages/EventDetailsWorkflowOperationDetails.js:
```diff
@@ -0,0 +1,176 @@
+                >
+                    {t("EVENTS.EVENTS.DETAILS.OPERATION_DETAILS.TITLE") /* Operation Details */}
+                </a>
+            </nav>
```
Is it possible to implement a navigation component for the hierarchy navigation and only hand over a list containing translation and sub-tab keys? Then this could be also used in the other workflow details components.

isabellakutger commented on modules/admin-ui-picard/app/src/components/events/partials/ModalTabsAndPages/EventDetailsWorkflowOperationDetails.js:
```diff
@@ -0,0 +1,176 @@
+                        <div className="obj-container">
+                            <table className="main-tbl">
+                                {isFetching || (
+                                    <>
```
`<tbody>`

isabellakutger commented on modules/admin-ui-picard/app/src/components/events/partials/ModalTabsAndPages/EventDetailsWorkflowOperationDetails.js:
```diff
@@ -0,0 +1,176 @@
+                                                {t("EVENTS.EVENTS.DETAILS.OPERATION_DETAILS.TABLE_HEADERS.FAIL_ON_ERROR") /* Fail on Error */ }
+                                            </td>
+                                            <td>{operationDetails.fail_on_error}</td>
+                                        </tr>
```
`</tbody>`

isabellakutger commented on modules/admin-ui-picard/app/src/components/events/partials/ModalTabsAndPages/EventDetailsWorkflowOperations.js:
```diff
@@ -0,0 +1,132 @@
+                                            <tr>
+                                                <th>{t("EVENTS.EVENTS.DETAILS.WORKFLOW_OPERATIONS.TABLE_HEADERS.STATUS") /* Status */}</th>
+                                                <th>{t("EVENTS.EVENTS.DETAILS.WORKFLOW_OPERATIONS.TABLE_HEADERS.TITLE") /* Title */}
+                                                    <i></i>
```
`<i/>`

isabellakutger commented on modules/admin-ui-picard/app/src/components/events/partials/ModalTabsAndPages/EventDetailsWorkflowOperations.js:
```diff
@@ -0,0 +1,132 @@
+                                                    <i></i>
+                                                </th>
+                                                <th>{t("EVENTS.EVENTS.DETAILS.WORKFLOW_OPERATIONS.TABLE_HEADERS.DESCRIPTION") /* Description */}
+                                                    <i></i>
```
`<i/>`

isabellakutger commented on modules/admin-ui-picard/app/src/thunks/eventDetailsThunks.js:
```diff
@@ -25,6 +25,19 @@ import {
+    loadEventWorkflowErrorsInProgress,
+    loadEventWorkflowErrorsSuccess,
+    loadEventWorkflowErrorsFailure,
+    loadEventWorkflowErrorDetailsInProgress, loadEventWorkflowErrorDetailsSuccess, loadEventWorkflowErrorDetailsFailure,
```
each in an own line

isabellakutger commented on modules/admin-ui-picard/app/src/thunks/eventDetailsThunks.js:
```diff
@@ -244,7 +257,22 @@ export const fetchWorkflows = (eventId) => async (dispatch, getState) => {
+    try {
+        dispatch(loadEventWorkflowDetailsInProgress());
+
+        const data = await axios.get(`admin-ng/event/${eventId}/workflows/${workflowId}.json`);
```
start URL in `axio.get()` with a slash otherwise the wrong URL will be used in the final opencast build

isabellakutger commented on modules/admin-ui-picard/app/src/thunks/eventDetailsThunks.js:
```diff
@@ -325,6 +353,64 @@ export const deleteWorkflow = (eventId, workflowId) => async (dispatch, getState
+        dispatch(loadEventWorkflowOperationsSuccess(workflowOperations));
+    } catch (e) {
+        dispatch(loadEventWorkflowOperationsFailure())
+        // todo!!!!!!!
```
Still a todo?

isabellakutger commented on modules/admin-ui-picard/app/src/thunks/eventDetailsThunks.js:
```diff
@@ -325,6 +353,64 @@ export const deleteWorkflow = (eventId, workflowId) => async (dispatch, getState
+        dispatch(loadEventWorkflowOperationDetailsSuccess(workflowOperationDetails));
+    } catch (e) {
+        dispatch(loadEventWorkflowOperationDetailsFailure())
+        // todo!!!!!!!
```
Still a todo?

isabellakutger commented on modules/admin-ui-picard/app/src/thunks/eventDetailsThunks.js:
```diff
@@ -325,6 +353,64 @@ export const deleteWorkflow = (eventId, workflowId) => async (dispatch, getState
+        dispatch(loadEventWorkflowErrorsSuccess(workflowErrors));
+    } catch (e) {
+        dispatch(loadEventWorkflowErrorsFailure())
+        // todo!!!!!!!
```
Still a todo?

isabellakutger reviewed (changes requested):
Some further things:

- In the console some warnings occur, when clicking through the hierachy. I think most of them are little fixes.
- the button on the workflow details page uses the wrong translation key
![Button](https://user-images.githubusercontent.com/34277297/150981881-f6921fb5-f734-47e9-bf08-ee49e5831fca.png)
- the notifications are not reset when choosing another event 
![Notification](https://user-images.githubusercontent.com/34277297/150982186-3fc0b3be-e8f7-4603-b0ad-9d790daddf99.png)

SimoneSchmidt committed:
Changes requested in PR #332

Added the changes requested for PR #332.

SimoneSchmidt committed:
Fixed style warnings

Fixed html nesting in hierarchy navigation for event details workflow details which caused warnings.

Commit:
Merge branch 'admin-ui-picard' into picard-workflow-details
